ddtrace/internal/datastreams/kombu.py

```diff
@@ -0,0 +1,46 @@
+from ddtrace import config
+from ddtrace.contrib.kombu.utils import HEADER_POS
+from ddtrace.contrib.kombu.utils import PUBLISH_BODY_IDX
+from ddtrace.contrib.kombu.utils import get_routing_key_from_args
+from ddtrace.contrib.kombu.utils import get_exchange_from_args
+from ddtrace.internal import core
+from ddtrace.internal.datastreams.processor import PROPAGATION_KEY
+from ddtrace.internal.datastreams.utils import _calculate_byte_size
+from ddtrace.internal.logger import get_logger
+
+
+log = get_logger(__name__)
+
+
+def handle_kombu_produce(args, kwargs, span):
+    from . import data_streams_processor as processor
+
+    dsm_identifier = get_exchange_from_args(args)
+    payload_size = 0
+    payload_size += _calculate_byte_size(args[HEADER_POS])
+    payload_size += _calculate_byte_size(args[PUBLISH_BODY_IDX])
+
+    has_routing_key = str(bool(get_routing_key_from_args(args))).lower()
+
+    pathway = processor().set_checkpoint(
+        ["direction:out", f"exchange:{dsm_identifier}", f"has_routing_key:{has_routing_key}", "type:rabbitmq"], payload_size=payload_size, span=span
+    )
+    encoded_pathway = pathway.encode()
+    args[HEADER_POS][PROPAGATION_KEY] = encoded_pathway
+
+
+def handle_kombu_consume(instance, message, span):
+    from . import data_streams_processor as processor
+
+    payload_size = 0
+    payload_size += _calculate_byte_size(message.body)
+    payload_size += _calculate_byte_size(message.headers)
+
+    ctx = processor().decode_pathway(message.headers.get(PROPAGATION_KEY, None))
+    queue = instance.queues[0].name
```

Comment:
@piochelepiotr 

"instance" is the consumer.  We're taking a guess here as to which channel the message came from, by assuming that the first channel of the consumer is correct.

Reply:
Also, this will return 'tasks' for the test

Reply:
we want to set `topic` to the queue name, not the channel name, but I believe it's what we are doing here?

Reply:
yes, this is the queue name.  The challenge here is that a channel may have more than one queue, and we're just picking the first

Reply:
should we check if len(queues) > 0? (it's probably always going to be the case, but we might get an exception if we don't check)